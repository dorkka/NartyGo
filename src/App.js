import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import PropTypes from 'prop-types';
import './App.css';
import ResortsData from './components/ResortsData';
import SpecificResortPage from './components/resortPage/SpecificResortPage';
import FooterLayout from './components/FooterLayout';
import TopLayout from './components/TopLayout';

const LayoutWrapper = props => (
  <div className="container-fluid App">
    <TopLayout />
    {props.children}
    <FooterLayout />
  </div>
);
class App extends Component {
  render() {
    return (
      <Router>
        <div>
          <Switch>
            <Route exact path="/" render={() => <LayoutWrapper><ResortsData /></LayoutWrapper>} />
            <Route exact path="/resorts" render={() => <LayoutWrapper><ResortsData /></LayoutWrapper>} />
            <Route
              path="/resorts/:id"
              render={props => <LayoutWrapper><SpecificResortPage {...props} /></LayoutWrapper>}
            />
            <Route render={() => <h3>404 page not found</h3>} />
          </Switch>
        </div>
      </Router>
    );
  }
}
LayoutWrapper.propTypes = {
  children: PropTypes.element.isRequired,
};

export default App;
