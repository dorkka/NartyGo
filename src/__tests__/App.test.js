import React from 'react';
import { render } from 'enzyme';
import { Provider } from 'react-redux';
import App from '../App';
import appStore from '../store/configureStore';

test('renders without crashing', () => {
  const wrapper = render(<Provider store={appStore}><App /></Provider>);
  expect(wrapper).toMatchSnapshot();
});
