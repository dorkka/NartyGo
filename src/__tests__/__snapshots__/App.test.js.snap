// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="container-fluid App"
  >
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-dark bg-dark"
      >
        <div
          class="collapse navbar-collapse"
          id="navbarColor02"
        >
          <ul
            class="navbar-nav mr-auto"
          >
            <li
              class="nav-item active"
            >
              <a
                class="nav-link"
                href="/"
              >
                Home 
                <span
                  class="sr-only"
                >
                  (current)
                </span>
              </a>
            </li>
          </ul>
          <ul
            class="nav navbar-nav navbar-right"
          >
            <li>
              <a
                class="nav-link"
                href="#"
              >
                Logowanie
              </a>
            </li>
          </ul>
        </div>
      </nav>
      <div
        class="jumbotron"
      >
        <h1
          class="display-4"
        >
          Witamy w serwisie NartyGo
        </h1>
        <p
          class="lead"
        >
          NartyGo to najlepszy serwis narciarski z najaktualniejszymi informacjami o ośrodkach narciarskich i warunkach na trasach!
        </p>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-3"
        >
          <p>
            Temperatura: 
          </p>
          <p>
            Zachmurzenie: 
          </p>
        </div>
        <div
          class="col-md-2"
        >
          <p>
            Sprawdź aktualną pogodę 
          </p>
          <p>
            <button
              class="btn btn-warning"
              type="button"
            >
              Sprawdź
            </button>
          </p>
        </div>
        <div
          class="col-md-7"
        >
          <p>
            Wyszukaj ośrodek
          </p>
          <input
            type="text"
          >
            
          </input>
          <p>
            <button
              class="btn btn-warning"
              type="button"
            >
              Szukaj
            </button>
          </p>
        </div>
      </div>
    </div>
    <div>
      There is no data
    </div>
    <div>
      <hr>
        
      </hr>
      <footer
        style="font-size:large;color:blue"
      >
        <p>
          © D-Softkol 2018 
        </p>
      </footer>
    </div>
  </div>
</div>
`;
